x-pack/plugins/data_enhanced/server/search/session/check_running_sessions.ts

```diff
@@ -125,10 +125,19 @@ function getSavedSearchSessionsPage$(
       sortOrder: 'asc',
       filter: nodeBuilder.or([
         nodeBuilder.and([
-          nodeBuilder.is(
-            `${SEARCH_SESSION_TYPE}.attributes.status`,
-            SearchSessionStatus.IN_PROGRESS.toString()
-          ),
+          nodeBuilder.or([
+            nodeBuilder.is(
+              `${SEARCH_SESSION_TYPE}.attributes.status`,
+              SearchSessionStatus.IN_PROGRESS.toString()
+            ),
+            nodeBuilder.and([
+              nodeBuilder.is(
+                `${SEARCH_SESSION_TYPE}.attributes.status`,
+                SearchSessionStatus.COMPLETE.toString()
+              ),
+              // TODO: check that attributes.expires < now
```

Comment:
@lizozom, @lukasolson I spent some time trying to understand how code behind `nodeBuilder` works to build a range query, but no luck. 
I think you have more experience in that part? Could you maybe help out?

Reply:
This would drastically  increase the amount of sessions we check every time!
This would pretty much mean we will keep fetching every session in the system, every 10 seconds, for a week.

I would instead run a separate task every 1 hour maybe to do less important updates.

And as for the management UI, we could just infer the `Expired` status on the FE (like we do with `expiresSoonWarning`)